src/api/app/controllers/concerns/webui/requests_filter.rb

```diff
@@ -42,4 +44,15 @@ def filter_by_involvement(filter_involvement)
       User.session.outgoing_requests
     end
   end
+
+  def filter_by_involvement_for_project(filter_by_involvement, project)
+    case filter_by_involvement
+    when 'all'
+      BsRequest.with_actions.where('bs_request_actions.target_project = (?) OR bs_request_actions.source_project = (?)', project.name, project.name)
```

Comment:
I worry that for projects like `openSUSE:Factory`, this query will take forever if we don't clamp it to just open requests by default. I don't know if there's an easy solution right now, but it's something to keep in mind for the future

Reply:
This query will not fire until it's results are used. And this usage includes pagination AFAICS...

Reply:
Right, pagination does save us here